lib/src/decorators/widget_decorators.dart

```diff
@@ -10,6 +10,45 @@ import '../core/decorator.dart';
 import '../factory/mix_provider_data.dart';
 import '../helpers/lerp_helpers.dart';
 
+class IntrinsicHeightDecorator
+    extends WidgetDecorator<IntrinsicHeightDecorator> {
+  const IntrinsicHeightDecorator({super.key});
+
+  @override
+  IntrinsicHeightDecorator lerp(
+    covariant Decorator<Decorator>? other,
```

Comment:
We should change the lerp to accept `IntrinsicHeightDecorator? other` without the covariant